### ISSUE TITLE
Getting access to a Frame from a Handle

### ISSUE THREAD

**Issue by woelper** [tesselode/kira#15]
Hi, and thanks for this great library!
I am experimenting with Kira and bevy, as well as building a super simple [audio player](https://github.com/woelper/lynx) for fun. I think it would be great to access a `Frame` at a time, for example to have a game entity react to the sound or add visualization to audio. I see that there is [get_frame_at_position](https://docs.rs/kira/0.5.3/kira/sound/struct.Sound.html#method.get_frame_at_position), but this is not implemented for the Handle. Would this be reasonable to add or am I missing a way to make this work with the current state maybe?

**Comment by tesselode** [comment]
After re-reading this issue, I thin what you were requesting is a way to see the frame a sound is currently outputting, right, @woelper ?

**Comment by woelper** [comment]
Thanks, yes, you are correct.

Sorry for derailing this a bit, but would this work with StreamingSounds too?

**Comment by tesselode** [comment]
Yes. While you still can't get the currently playing frame from a handle, you can create an effect that sends the frames (or information about the frames, like the average amplitude) to an effect handle. I may do a write-up on how to implement this (or maybe even add it as a default effect).

**Comment by jakerr** [comment]
> you can create an effect that sends the frames (or information about the frames, like the average amplitude) to an effect handle.

I made a demo that does exactly this in my kira wrapper for Bevy.

See: 
https://github.com/jakerr/bevy_mod_kira/blob/main/examples/effects/level_monitor/mod.rs
and it's usage
https://github.com/jakerr/bevy_mod_kira/blob/main/examples/level_monitor.rs

I realize it's been a year since the last comment here but thought for posterity's sake I'd share.
